runtime/src/main/scala/akka/grpc/internal/NettyClientUtils.scala

```diff
@@ -25,21 +27,28 @@ object NettyClientUtils {
    * INTERNAL API
    */
   @InternalApi
-  def createChannel(settings: GrpcClientSettings): ManagedChannel = {
-    var builder =
-      NettyChannelBuilder
-        .forAddress(settings.host, settings.port)
-        .flowControlWindow(NettyChannelBuilder.DEFAULT_FLOW_CONTROL_WINDOW)
-
-    builder = settings.trustedCaCertificate.map(c => GrpcSslContexts.forClient.trustManager(loadCert(c)).build)
-      .map(ctx => builder.negotiationType(NegotiationType.TLS).sslContext(ctx))
-      .getOrElse(builder.negotiationType(NegotiationType.PLAINTEXT))
-
-    builder = settings.overrideAuthority.map(builder.overrideAuthority(_)).getOrElse(builder)
-
-    builder = settings.userAgent.map(builder.userAgent(_)).getOrElse(builder)
-
-    builder.build
+  def createChannel(settings: GrpcClientSettings)(implicit ec: ExecutionContext): Future[ManagedChannel] = {
```

Comment:
How does it work now when the connection/channel is disconnected (e.g. server stopped)? All running streams will fail and new calls will fail?

Also, what happens after `client.close()` (`channel.shutdown()`) will that fail new calls immediately or do we have to add some fail fast for that case.

Shall we expose something like `def closed: Future[Done]`? Then we can complete that in `close()` and also complete it when the channel is disconnected. Assuming that we could listen to such even in the `channel`.

Then one can at least act on this and restart the client and thereby do a new discovery. Poor mans retry/reconnect until we have better things built in.

Wdyt?

Reply:
Since it's a `ManagedChannel` it will try to reconnect, but of course even then it won't run our service lookup anymore.

If we can listen to the `channel` disconnecting I think we should replace the `channel` 'inside' the client rather than leaving this up to the API user.